src/metadata/trackmetadata.cpp

```diff
@@ -0,0 +1,121 @@
+#include "metadata/trackmetadata.h"
+
+#include "util/math.h"
+
+namespace Mixxx {
+
+/*static*/ const double TrackMetadata::BPM_UNDEFINED = 0.0;
+/*static*/ const double TrackMetadata::BPM_MIN = 0.0; // lower bound (inclusive)
+/*static*/ const double TrackMetadata::BPM_MAX = 300.0; // lower bound (inclusive)
+
+/*static*/ const float TrackMetadata::REPLAYGAIN_UNDEFINED = 0.0f;
+/*static*/ const float TrackMetadata::REPLAYGAIN_MIN = 0.0f; // lower bound (inclusive)
+/*static*/ const float TrackMetadata::REPLAYGAIN_0DB = 1.0f;
+
+namespace {
+
+double parseBpmString(const QString& sBpm, bool* pValid = 0) {
+    if (pValid) {
+        *pValid = false;
+    }
+    if (sBpm.trimmed().isEmpty()) {
+        return TrackMetadata::BPM_UNDEFINED;
+    }
+    bool bpmValid = false;
+    double bpm = sBpm.toDouble(&bpmValid);
+    if (bpmValid) {
+        if (TrackMetadata::BPM_UNDEFINED == bpm) {
+            // special case
+            if (pValid) {
+                *pValid = true;
+            }
+            return bpm;
+        }
+        while (TrackMetadata::BPM_MAX < bpm) {
+            // TODO(XXX): Why?
```

Comment:
Comment Proposal: 

// TBPM is an integer in http://id3.org/id3v2.3.0
// There are probably App that store the BPM with decimal points or 
// multiplied by 10 or 100. 
// A value of 1283 bpm can't be a valid so we assume 128.3 bpm.
// This was introduced with   
// https://github.com/mixxxdj/mixxx/commit/77c698f656992f0a1c384806e9948dee523a6190
// TODO(XXX) verify that